### PR TITLE
Implement mask functionality in the dxDateBox 🎭

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

**Review by jsdmitry — DISMISSED** [review]

File: js/localization/ldml/date.parser.js [hunk 1]
```diff
@@ -114,7 +115,27 @@ var PATTERN_PARSERS = {
...
+    s: "getSeconds",
+    S: "getMilliseconds"
+};
+
 var PATTERN_SETTERS = {
+    E: function(date, value) {
+        if(value < 0) {
+            return;
+        }
+        date.setDate(date.getDate() - date.getDay() + value % 6);
```
Comment (vconst) [inline js/localization/ldml/date.parser.js]: Remove normalization (value % 6)

File: js/localization/ldml/date.parser.js [hunk 2]
```diff
@@ -190,9 +211,22 @@ var getRegExpInfo = function(format, dateParts) {
     };
 };
 
+var getPatternSetter = function(patternChar) {
+    return PATTERN_SETTERS[patternChar] || noop;
+};
+
+var getPatternGetter = function(patternChar) {
```
Comment (vconst) [inline js/localization/ldml/date.parser.js]: There are tests for this method?

File: js/localization/ldml/date.parser.js [hunk 3]
```diff
@@ -159,8 +180,6 @@ var getRegExpInfo = function(format, dateParts) {
         isEscaping,
         patterns = [];
 
-    format = escapeRegExp(format);
```
Comment (vconst) [inline js/localization/ldml/date.parser.js:162]: Why is this line moved?

File: js/localization/ldml/date.parser.js [hunk 4]
```diff
@@ -181,7 +200,9 @@ var getRegExpInfo = function(format, dateParts) {
             regexpText += "(" + regexpPart(count, dateParts) + ")";
             i += count - 1;
         } else {
-            regexpText += char;
+            char = escapeRegExp(char);
+            patterns.push(char);
```
Comment (vconst) [inline js/localization/ldml/date.parser.js:182]: How you differ pattern "y" and separator "y" if patterns[i] for these cases are equal? Maybe, a better way to introduce a special pattern symbol for separators?

File: js/localization/ldml/date.parser.js (hunk 1 above)
Comment (vconst) [reply js/localization/ldml/date.parser.js]: Where are tests for this new setter?

**Review by vconst — DISMISSED** [review]

File: js/localization/ldml/date.parser.js (hunk 4 above)
Comment (mrjohn0011) [reply js/localization/ldml/date.parser.js:182]: You can escape this letter: `'year: 'yyyy`

**Review by jsdmitry — APPROVED** [review]
